### PR TITLE
Add changes detection to record editing

### LINKED ISSUE #1360: Add change detection for record editing
Similar to resource editing change detection, also add it for record creation/edit
This is referring to the "Unsaved changes warning"

### PR DESCRIPTION
**Pull request by kelanik8**
# The following changes are implemented
TODO: Summary

# Changes in the user interface:
TODO: Add screenshots, recordings or remove this section

# Checklist when submitting a final (!draft) PR
 - [ ] Commits are tidied up, squashed if needed and follow guidelines in CONTRIBUTING.md
 - [ ] Code builds
 - [ ] All existing tests pass
 - [ ] All new critical code is covered by tests
 - [ ] PR is linked to the relevant issue(s)
 - [ ] Rebased with the target branch

### CODE REVIEW

**Review by Fajfa — COMMENTED** [review]
Regarding this, please test it how it works with modals. 
If all is good, move to QC so it can be confirmed.

Also rebase with 2023.3.x asap

**Review by Fajfa — APPROVED** [review]

**Comment by katrinDY** [comment]
A couple of issues:
- in RL click on the dropdown menu for a record -> edit -> don't make any changes -> go back and the unsaved message is shown => *not longer an issue*
-  in RL click on the dropdown menu for a record -> edit -> don't make any changes -> click on save -> see the toast message `Could not update record: Cannot read properties of undefined (reading 'recordID')`

- open a record in modal -> edit a field -> ctr + z the change -> close the modal -> the unsaved message is shown (tried this on text, select and number type of field; the issue should be present for the rest of the field types) => *Happens only on CRM not to new ns* *BE removes HTML and prevents change detection to work - ticket to be added for this by Tolu*
- click on add btn in RL -> add value to a field -> go back -> the unsaved message is *not* shown => *won't be fixed since we're removing change detection when a resource sys created*
- open record in a modal -> click on add -> add value to a field -> close the modal -> the unsaved message is *not* shown =>  *won't be fixed since we're removing change detection when a resource sys created*

**Review by Fajfa — CHANGES_REQUESTED** [review]
Would it make sense to keep using the .clone(), but adjusting it on the class
So we can always use clone, similar to how you do it with other change detection

**Review by Fajfa — APPROVED** [review]

**Comment by katrinDY** [comment]
One last thing and we should be good to go: *issue is present only in CRM and Case Management ns*

**Review by Fajfa — APPROVED** [review]

**Comment by katrinDY** [comment]
The issue is still present

**Review by Fajfa — APPROVED** [review]

**Review by Fajfa — APPROVED** [review]

**Comment by katrinDY** [comment]
- happens only in CRM and Case management

File: client/web/compose/src/views/Public/Pages/Records/View.vue [hunk 1]
```diff
@@ -640,6 +640,7 @@ export default {
     checkUnsavedChangesOnModal (bvEvent, modalId) {
       if (modalId === 'record-modal' && !this.inCreating) {
         const recordStateChange = this.compareRecordValues() ? window.confirm(this.$t('general:editor.unsavedChanges')) : true
+        console.log('recordStateChange', recordStateChange)
```
Comment (katrinDY) [inline client/web/compose/src/views/Public/Pages/Records/View.vue]: Do you need this? If not, please, remove

**Review by Fajfa — APPROVED** [review]

**Pull request merged**